airflow-core/src/airflow/api_fastapi/app.py

```diff
@@ -93,6 +94,7 @@ def create_app(apps: str = "all") -> FastAPI:
         init_plugins(app)
         init_auth_manager(app)
         init_flask_plugins(app)
+        init_ui_plugins(app)
```

Comment:
Init UI plugins at api server start time, this way the warning is displayed early. (without this it would be displayed when accessing the first time the plugins endpoint)

Keep the factory pattern.

Reply:
Big DevEx improvement right there, imo